lsp-mode.el

```diff
@@ -479,46 +479,64 @@ the server has requested that."
-                                    "[/\\\\]\\.hg\\'"
-                                    "[/\\\\]\\.bzr\\'"
-                                    "[/\\\\]_darcs\\'"
-                                    "[/\\\\]\\.svn\\'"
-                                    "[/\\\\]_FOSSIL_\\'"
-                                    ;; IDE or build tools
-                                    "[/\\\\]\\.idea\\'"
-                                    "[/\\\\]\\.ensime_cache\\'"
-                                    "[/\\\\]\\.eunit\\'"
-                                    "[/\\\\]node_modules"
-                                    "[/\\\\]\\.fslckout\\'"
-                                    "[/\\\\]\\.tox\\'"
-                                    "[/\\\\]dist\\'"
-                                    "[/\\\\]dist-newstyle\\'"
-                                    "[/\\\\]\\.stack-work\\'"
-                                    "[/\\\\]\\.bloop\\'"
-                                    "[/\\\\]\\.metals\\'"
-                                    "[/\\\\]target\\'"
-                                    "[/\\\\]\\.ccls-cache\\'"
-                                    "[/\\\\]\\.vscode\\'"
-                                    ;; Autotools output
-                                    "[/\\\\]\\.deps\\'"
-                                    "[/\\\\]build-aux\\'"
-                                    "[/\\\\]autom4te.cache\\'"
-                                    "[/\\\\]\\.reference\\'"
-                                    ;; .Net Core build-output
-                                    "[/\\\\]bin/Debug\\'"
-                                    "[/\\\\]obj\\'")
+(define-obsolete-variable-alias 'lsp-file-watch-ignored 'lsp-file-watch-ignored-directories "7.1.0")
+
+(defcustom lsp-file-watch-ignored-directories
+  '(; SCM tools
+    "[/\\\\]\\.git\\'"
+    "[/\\\\]\\.hg\\'"
+    "[/\\\\]\\.bzr\\'"
+    "[/\\\\]_darcs\\'"
+    "[/\\\\]\\.svn\\'"
+    "[/\\\\]_FOSSIL_\\'"
+    ;; IDE or build tools
+    "[/\\\\]\\.idea\\'"
+    "[/\\\\]\\.ensime_cache\\'"
+    "[/\\\\]\\.eunit\\'"
+    "[/\\\\]node_modules"
+    "[/\\\\]\\.fslckout\\'"
+    "[/\\\\]\\.tox\\'"
+    "[/\\\\]dist\\'"
+    "[/\\\\]dist-newstyle\\'"
+    "[/\\\\]\\.stack-work\\'"
+    "[/\\\\]\\.bloop\\'"
+    "[/\\\\]\\.metals\\'"
+    "[/\\\\]target\\'"
+    "[/\\\\]\\.ccls-cache\\'"
+    "[/\\\\]\\.vscode\\'"
+    ;; Autotools output
+    "[/\\\\]\\.deps\\'"
+    "[/\\\\]build-aux\\'"
+    "[/\\\\]autom4te.cache\\'"
+    "[/\\\\]\\.reference\\'"
+    ;; .Net Core build-output
+    "[/\\\\]bin/Debug\\'"
+    "[/\\\\]obj\\'")
   "List of regexps matching directory paths which won't be monitored when creating file watches."
   :group 'lsp-mode
   :type '(repeat string)
-  :package-version '(lsp-mode . "6.1"))
-
-(defun lsp-file-watch-ignored ()
-  lsp-file-watch-ignored)
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defun lsp-file-watch-ignored-directories ()
+  lsp-file-watch-ignored-directories)
```

Comment:
the idea of this function was to allow the users to add advice over it and alter the lsp-file-watch-ignored dynamically. I am not 100% if this solution is still relevant(it was requested and implemented a long time ago) but we may keep the old function as alias at least or rever that rename.